### PR TITLE
Fix close button change on delete in simplemde area

### PR DESCRIPTION
**Pull request by zeripath**
Fix issue with close button changing when deleting in the simplemde textarea.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: web_src/js/index.js [hunk 1]
```diff
@@ -1184,6 +1184,12 @@ async function initRepository() {
     // Change status
     const $statusButton = $('#status-button');
     $('#comment-form textarea').on('keyup', function () {
+      const $simplemde = $(this).data('simplemde');
+      if ($simplemde && $simplemde.value().length !== 0) {
+        $statusButton.text($statusButton.data('status-and-comment'));
+        return;
+      }
+
```
Comment (silverwind) [inline web_src/js/index.js]: Try this more compact solution for this handler:

```js
const $simplemde = $(this).data('simplemde');
const value = ($simplemde && $simplemde.value()) ? $simplemde.value() : $(this).val();
$statusButton.text($statusButton.data(value.length === 0 ? 'status' : 'status-and-comment'));
```

**Review by 6543 — APPROVED** [review]
works :)

File: web_src/js/index.js (hunk 1 above)
Comment (6543) [reply web_src/js/index.js]: tested, didnt work ...?

Comment (silverwind) [reply web_src/js/index.js]: Hmm maybe this:

```js
const $simplemde = $(this).data('simplemde');
const value = ($simplemde && $simplemde.value()?.length) ? $simplemde.value() : $(this).val();
$statusButton.text($statusButton.data(value.length === 0 ? 'status' : 'status-and-comment'));
```

Comment (6543) [reply web_src/js/index.js]: @silverwind you are right, the first solution works well, if you replace the code below too ...

Comment (6543) [reply web_src/js/index.js]: sorry ... didn't read the whole func and wat it does until now :swan:

**Review by silverwind — APPROVED** [review]

File: web_src/js/index.js (hunk 1 above)
Comment (silverwind) [reply web_src/js/index.js]: Of course I meant the whole function, that's why I wrote "this handler" 😉

**Comment by zeripath** [comment]
Make lgtm work

**Pull request merged**